hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/mongodb/UpdateProcessorSupplier.java

```diff
@@ -0,0 +1,150 @@
+    }
+
+    @Nonnull
+    @Override
+    public Collection<? extends Processor> get(int count) {
+        Processor[] processors = new Processor[count];
+
+        for (int i = 0; i < count; i++) {
+            Processor processor = new WriteMongoP<>(
+                    new WriteMongoParams<Document>()
+                            .setClientSupplier(() -> MongoClients.create(connectionString))
+                            .setDatabaseName(databaseName)
+                            .setCollectionName(collectionName)
+                            .setDocumentType(Document.class)
+                            .setDocumentIdentityFn(doc -> doc.getObjectId("_id"))
+                            .setDocumentIdentityFieldName("_id")
+                            .setCommitRetryStrategy(DEFAULT_COMMIT_RETRY_STRATEGY)
+                            .setTransactionOptionsSup(() -> DEFAULT_TRANSACTION_OPTION)
+                            .setIntermediateMappingFn(this::rowToUpdateDoc)
+                            .setWriteMode(WriteMode.UPDATE_ONLY)
+                            .setWriteModelFn(this::write)
+            );
+
+            processors[i] = processor;
+        }
+        return asList(processors);
+    }
+
+    private WriteModel<Document> write(Document row) {
+        List<Bson> updateToPerform = new ArrayList<>();
+        for (int i = 0; i < fieldNames.size(); i++) {
+            String fieldName = fieldNames.get(i);
+            Object updateExpr = updates.get(i);
+            if (updateExpr instanceof Bson) {
+                Document document = Document.parse(((Bson) updateExpr)
+                                            .toBsonDocument(Document.class, defaultCodecRegistry()).toJson());
+                ParameterReplacer.replacePlaceholders(document, evalContext);
+                updateExpr = document;
+            } else if (updateExpr instanceof DynamicParameter) {
+                updateExpr = evalContext.getArgument(((DynamicParameter) updateExpr).getIndex());
+            }
+
+            updateToPerform.add(Updates.set(fieldName, updateExpr));
+        }
+
+        List<Bson> all = new ArrayList<>();
+        for (String field : pkFields) {
+            all.add(Filters.eq(field, row.get(field)));
+        }
+        Bson filter = Filters.and(all);
+        return new UpdateOneModel<>(filter, updateToPerform);
+    }
+
+    /**
+     * Row parameter contains values for PK fields.
+     *
+     * In Mongo we - for now - assume only default {@code _id} field, as it's
+     * in the {@linkplain MongoSqlConnectorBase#getPrimaryKey(Table)}.
+     */
+    private Document rowToUpdateDoc(JetSqlRow row) {
+        Object[] values = row.getValues();
+
+        Document doc = new Document();
+        int index = 0;
+        for (String pkField : pkFields)  {
+            Object value = values[index++];
+            if (value instanceof String) {
+                value = new ObjectId((String) value);
+            }
```

Comment:
I've changed it, so that default type is OBJECT -> no coercion to String is done in Select processor. This way whatever comes from Mongo is passed here